theano/sandbox/gpuarray/blas.py

```diff
@@ -116,11 +96,11 @@ class GpuGemm(BlasOp, Gemm):
     _f16_ok = True
 
     def make_node(self, C, alpha, A, B, beta):
-        alpha = as_tensor_variable(alpha)
-        beta = as_tensor_variable(beta)
-        A = as_gpuarray_variable(A)
-        B = as_gpuarray_variable(B)
-        C = as_gpuarray_variable(C)
+        ctx_name = infer_context_name(C, A, B)
+        Gemm.make_node(self, C, alpha, A, B, beta)
```

Comment:
Remove `Gemm.make_node`, and perform the checks about dtype and ndim in this function. I don't think the check about overlapping memory is needed.
Also re-add `as_tensor_variable` on `alpha` and `beta`.